src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -578,8 +578,9 @@ public boolean registerSubscriber(HttpBodySubscriberWrapper<?> subscriber) {
                         if (debug.on()) {
                             debug.log("body subscriber registered: " + count);
                         }
+                        return true;
                     }
-                    return true;
+                    return false;
                 }
```

Comment:
Is this change actually part of the fix here, or something else that needed to be fixed?

Reply:
Probably both. I was looking at why the subscriber didn't get properly unregistered and stumbled on this small issue. Tests showed that it wasn't sufficient to fix the issue so I continued debugging.
The semantics of the method is that it should return `true` only if the subscriber was added to the list, so it really should have returned `false` here.